### DISCUSSION TITLE
Zapier Integration for Automatic Assignment of Access to Password-Protected Series/Chapters

### DISCUSSION THREAD

**Discussion by bravking** [Tetrakern/fictioneer#40] [General]
Previously, I've been using Zapier to assign custom Wordpress roles to users after a new Stripe/Paypal subscription payment. 

But, I've been wanting to assign automatic access to password-protected series/chapters based on plans they're subscribed to using Zapier's Wordpress API, hence the question below:

_How do I assign specific password-protected series and chapters' access automatically using Zapier's Wordpress API? What are the specific usermeta values I need to use? And, is this even feasible?_

<img width="319" alt="image" src="https://github.com/Tetrakern/fictioneer/assets/161727478/3bd41e0b-dc72-4ee6-a7dc-3fb4e1552e34">

**Comment by Tetrakern** [comment]
I'm not familiar with Zapier, so I cannot say anything about their API. If I understand your question correctly, you want to automatically unlock protected posts for user fulfilling certain criteria? The user meta you are looking for is `fictioneer_post_unlocks`, which is an array of post IDs (cast to string) or null if empty. So technically, you would need to add the ID of the post to unlock to this array (and make sure the array exists).

**Reply by bravking** [reply]
ah. got it. so if there's a bunch of locked chapters under particular story, does that mean i need to list out all the post ID's of all the locked chapters? Or is there a way for me to only list the story but unlock all the chapters for that particular story?

**Reply by Tetrakern** [reply]
Unlocking the story will also unlock all associated chapters, regardless of whether the story is actually locked.
